examples/siro_sandbox/sandbox_app.py

```diff
@@ -140,16 +160,49 @@ def __init__(self, args, config, gui_input):
         self._cursor_style = None
         self._can_grasp_place_threshold = args.can_grasp_place_threshold
 
-        # not sure if these are needed here for linting
-        # self._held_target_obj_idx = None
-        # self._num_remaining_objects = None
-        # self._num_busy_objects = None
-        # self._target_obj_ids = None
-        # self._goal_positions = None
+        self.reset_environment()
+
+    def find_episode_save_filepath_base(self, session_filepath_base):
+        retval = (
+            self._save_filepath_base + "." + str(self._num_recorded_episodes)
+        )
+        self._num_recorded_episodes += 1
+        return retval
+
+    def save_episode_recorder_dict(self):
+        if not self._save_filepath_base or not len(self._step_recorder._steps):
+            return
+
+        filepath_base = self.find_episode_save_filepath_base(
+            self._save_filepath_base
+        )
+
+        json_filepath = filepath_base + ".json.gz"
+        save_as_json_gzip(self._episode_recorder_dict, json_filepath)
+
+        pkl_filepath = filepath_base + ".pkl.gz"
+        save_as_pickle_gzip(self._episode_recorder_dict, pkl_filepath)
 
-        self.on_environment_reset()
+    def start_episode_recorder(self):
+        assert self._save_filepath_base and self._step_recorder
+        ep_dict: Any = dict()
+        ep_dict["start_time"] = datetime.now()
+        ep_dict["dataset"] = self._dataset
+        ep_dict["episode_id"] = self.env.current_episode.episode_id
 
-    def on_environment_reset(self):
+        ep_dict["target_obj_ids"] = self._target_obj_ids
```

Comment:
This mapping will be helpful, e.g., because `grasp_mgr_snap_idx` records the sim-object id of the held object.